### PR TITLE
[18.0][MIG] delivery_carrier_warehouse: Migration to 18.0

### PR DESCRIPTION
**Pull request by chaule97**
- Since the `delivery` module in [version 15.0](https://github.com/odoo/odoo/pull/110686/files) depends on `sale` instead of `sale_stock`, I need to update the `delivery_carrier_warehouse` module to depend on `stock_delivery` instead of `delivery`.

### CODE REVIEW

File: delivery_carrier_warehouse/models/stock_warehouse.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class StockWarehouse(models.Model):
+    _inherit = "stock.warehouse"
+
+    delivery_carrier_id = fields.Many2one(
+        "delivery.carrier",
+        string="Delivery Method",
+        help="Default delivery method used in sales orders. Will be applied "
```
Comment (ivantodorovich) [inline delivery_carrier_warehouse/models/stock_warehouse.py:13]: Could you add `check_company=True` here?

File: delivery_carrier_warehouse/views/stock_warehouse.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<odoo>
+    <record id="view_warehouse_inherit" model="ir.ui.view">
+        <field name="name">stock.warehouse.inherit</field>
+        <field name="model">stock.warehouse</field>
+        <field name="inherit_id" ref="stock.view_warehouse" />
+        <field name="arch" type="xml">
+            <field name="partner_id" position="after">
+                <field
+                    name="delivery_carrier_id"
+                    domain="[('company_id', 'in', (False, company_id))]"
```
Comment (ivantodorovich) [inline delivery_carrier_warehouse/views/stock_warehouse.xml]: Normally, if you use `check_company=True` on the field attribute, the domain is automatically set by the ORM

**Review by ivantodorovich — APPROVED** [review]

File: delivery_carrier_warehouse/views/stock_warehouse.xml (hunk 2 above)
Comment (chaule97) [reply delivery_carrier_warehouse/views/stock_warehouse.xml]: thanks for your knowledges

File: delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py [hunk 3]
```diff
@@ -1,13 +1,14 @@
 # Copyright 2020 Camptocamp SA
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
-from odoo.tests.common import Form, TransactionCase
+from odoo.tests import Form
 
+from odoo.addons.base.tests.common import BaseCommon
 
-class TestSaleDeliveryCarrierPreference(TransactionCase):
+
+class TestSaleDeliveryCarrierPreference(BaseCommon):
```
Comment (jbaudoux) [inline delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py]: Why you changed `TransactionCase` to `BaseCommon` ?

The class name `TestSaleDeliveryCarrierPreference` looks like a wrong copy/paste

Comment (chaule97) [reply delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py]: > Why you changed TransactionCase to BaseCommon ?

[BaseCommon](https://github.com/odoo/odoo/blob/2fe55ae592e5812c5ee5e39ed1e8332bde608cb2/odoo/addons/base/tests/common.py#L20) inherrite TransactionCase, and we don't need to write `cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))`

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
I was a bit too quick merging this. I'm moving this module to delivery-carrier repo. See https://github.com/OCA/wms/pull/978

**Pull request merged**